### PR TITLE
ROCm changes for PyT2.0 support

### PR DESCRIPTION
**Pull request by jataylo**
- Removed warnings on fallback to aten.miopen_batch_norm (due to perf diff)

- Refactored ROCm skips in opinfo

- sync with some upstream changes to remove install_triton,

### CODE REVIEW

File: tools/dynamo/verify_dynamo.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def get_rocm_version():
 def check_cuda():
     import torch
 
-    if not torch.cuda.is_available():
+    if not torch.cuda.is_available() or torch.version.cuda is None:
```
Comment (dllehr-amd) [inline tools/dynamo/verify_dynamo.py:84]: Did we plan to float this as a standalone PR?

File: torch/_inductor/lowering.py [hunk 2]
```diff
@@ -1403,6 +1403,9 @@ def apply_constraint(arg, fx_arg):
 make_fallback(aten.zeros.names)
 
 
+# ROCm specific
+make_fallback(aten.miopen_batch_norm, warn=False)
```
Comment (dllehr-amd) [inline torch/_inductor/lowering.py:1407]: We where going to hold off on this correct?

File: test/inductor/test_torchinductor.py [hunk 3]
```diff
@@ -3013,8 +3013,6 @@ def fn(w, i, o):
             (torch.randn([10, 4]), torch.randint(10, [8]), torch.tensor([0, 2, 6])),
         )
 
-    # FIXME: https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/3597
-    @skipIfRocm
     def test_batch_norm_2d(self):
```
Comment (dllehr-amd) [inline test/inductor/test_torchinductor.py:3016]: If we don't upload the miopen batch norm, this would stay skipped correct?

File: tools/dynamo/verify_dynamo.py (hunk 1 above)
Comment (jataylo) [reply tools/dynamo/verify_dynamo.py:84]: That PR is up now
https://github.com/pytorch/pytorch/pull/97013

File: .ci/pytorch/common_utils.sh [hunk 4]
```diff
@@ -149,38 +149,6 @@ function clone_pytorch_xla() {
   fi
 }
 
-function install_filelock() {
```
Comment (dllehr-amd) [inline .ci/pytorch/common_utils.sh:152]: This looks to be cleaning up what was already done in 72b9d45e76872f65a8de1cb1a61870739a97407a? 
Our latest commit 18848a1d64c97bb53011536ce5f8520c690189a0 seems to have accidently brought this along.  It looks good, and I can't see any other remnants of the previous commit in here.

**Review by dllehr-amd — APPROVED** [review]
LGTM!

File: torch/_inductor/lowering.py (hunk 2 above)
Comment (jataylo) [reply torch/_inductor/lowering.py:1407]: This change isn't to register miopen_batch_norm as a decomp as we tried earlier, this instead passes the miopen_batch_norm through the `register_lowering` logic to create the fallback for aten.miopen_batch_norm in the same way unsupported ops have been dealt with in the missing ops tracker https://github.com/pytorch/pytorch/issues/93757
https://github.com/pytorch/pytorch/blob/master/torch/_inductor/lowering.py#L1043

After making this change the generate codes will still to aten.miopen_batch_norm instead of lowering, but will not output this as a warning in the debug log and allows some UTs to pass.

Previously in `test_torchinductor` the batch_norm UTs would fail due to missing lowering
```
torch._dynamo.exc.BackendCompilerFailed: backend='compile_fx_wrapper' raised:
MissingOperatorWithoutDecomp: missing lowering
  target: aten.miopen_batch_norm.default
  args[0]: TensorBox(StorageBox(
    InputBuffer(name='primals_6', layout=FixedLayout('cuda', torch.float32, size=[2, 10, 8, 8], stride=[640, 64, 8, 1]))
  ))
  args[1]: TensorBox(StorageBox(
    InputBuffer(name='primals_1', layout=FixedLayout('cuda', torch.float32, size=[10], stride=[1]))
  ))
  args[2]: TensorBox(StorageBox(
    InputBuffer(name='primals_2', layout=FixedLayout('cuda', torch.float32, size=[10], stride=[1]))
  ))
  args[3]: TensorBox(StorageBox(
    InputBuffer(name='primals_3', layout=FixedLayout('cuda', torch.float32, size=[10], stride=[1]))
  ))
  args[4]: TensorBox(StorageBox(
    InputBuffer(name='primals_4', layout=FixedLayout('cuda', torch.float32, size=[10], stride=[1]))
  ))
  args[5]: False
  args[6]: 0.1
  args[7]: 1e-05
```

At the moment the reason for this passing is not well understood, it could be some additional logic that is added in the make_fallback that is required for lowering in some cases e.g. https://github.com/pytorch/pytorch/blob/master/torch/_inductor/lowering.py#L1040

This needs more investigation and a perf comparison to see if this change will reduce the overhead of creating a fallback during the model run.

File: test/inductor/test_torchinductor.py (hunk 3 above)
Comment (jataylo) [reply test/inductor/test_torchinductor.py:3016]: When the fallback is created in `lowering.py` this UT will pass. Needs some investigation into why this is the case. (Details in above comment)
